WireDomain/Sources/WireDomain/Repositories/Team/TeamRepository.swift

```diff
@@ -129,6 +135,56 @@ final class TeamRepository: TeamRepositoryProtocol {
         }
     }
 
+    public func pullSelfTeamRoles() async throws {
+        let teamRoles = try await fetchSelfTeamRolesRemotely()
+        try await storeTeamRolesLocally(teamRoles)
+    }
+
+    public func pullSelfTeamMembers() async throws {
+        let teamMembers = try await fetchSelfTeamMembersRemotely()
+        try await storeTeamMembersLocally(teamMembers)
+    }
+
+    public func pullSelfLegalHoldStatus() async throws {
+        let selfUser = await context.perform { [userRepository] in
+            userRepository.fetchSelfUser()
+        }
+
+        let selfUserLegalHold = try await fetchSelfLegalhold()
+
+        switch selfUserLegalHold.status {
+        case .pending:
+            guard let selfClientID = selfUser.selfClient()?.remoteIdentifier else {
```

Comment:
I think there's coredata threading violation, maybe WireDomain has not `-com.apple.CoreData.ConcurrencyDebug 1` set in its scheme.

Reply:
is it in the scheme?  and if not can you add it ?